### ISSUE TITLE
pagination issue

### ISSUE THREAD

**Issue by mkraetke** [transpect/CoCoTeX#23]
As we noticed in an [internal project](https://redmine.le-tex.de/issues/18033), for headings whose components (author, affiliation etc.) extend over two pages, the page number in the table of contents does not reference the page with the beginning of the heading environment, but rather its end.

**Comment by LupinoTech** [comment]
Found a way to mitigate the problem minimally invasive. Stylesheet maintainers need to  get active, thou, to incorporate the new `\tpGenerateToC` macro into their styles.

I'm goint to build something similar in the `ally` branch, as well.

**Comment by mkraetke** [comment]
Thank you. I assume that we need to insert `\tpGenerateToC` before the actual heading?

**Comment by LupinoTech** [comment]
No, that is something that needs to go in the .sty, not in the .tex file.
